pkg/cmd/infra/router/template.go

```diff
@@ -393,6 +471,18 @@ func (o *TemplateRouterOptions) Complete() error {
 	}
 	o.CaptureHTTPResponseHeaders = captureHTTPResponseHeaders
 
+	httpResponseHeaders, err := parseHeadersToBeSetOrDeleted(o.HTTPResponseHeadersString)
+	if err != nil {
+		return err
+	}
+	o.HTTPResponseHeaders = httpResponseHeaders
```

Comment:
`httpResponseHeaders` can be nil here, do you still want to set the `o.HTTPResponseHeaders` to nil if so?